src/metabase/types.clj

```diff
@@ -5,6 +5,7 @@
 (derive :type/Dictionary :type/Collection)
 (derive :type/Array :type/Collection)
 
+(derive :type/Nil :type/*)
```

Comment:
I "need" it for multimethod dispatch. It's used implicitly everywhere I'm dispatching on `:type/*`.

I think in terms of mental/conceptual overhead this is no worse than redoubling Number to :type/Number.